src/time_utils.c

```diff
@@ -41,6 +41,45 @@ subtract_interval_from_now(Oid timetype, const Interval *interval)
 	return res;
 }
 
+Datum
+ts_time_datum_convert_arg(Datum arg, Oid *argtype, Oid timetype)
+{
+	Oid type = *argtype;
+
+	if (!OidIsValid(type) || type == UNKNOWNOID)
+	{
+		Oid infuncid = InvalidOid;
+		Oid typeioparam;
+
+		type = timetype;
+		getTypeInputInfo(type, &infuncid, &typeioparam);
+
+		switch (get_func_nargs(infuncid))
+		{
+			case 1:
+				/* Functions that take one input argument, e.g., the Date function */
+				arg = OidFunctionCall1(infuncid, arg);
```

Comment:
Am I correct that this is a reuse of variable `arg`, which first contains the original argument, and then assigned the converted argument? I suggest to use different variable and actually it will contain different object to my understanding.